### PR TITLE
Connor/eatery db

### PR DESCRIPTION
**Pull request by connorreinhold**
- Add reading and writing db state
- Add item parsing system into menus from db
- Add event schedules for day of week

### CODE REVIEW

File: api/datatype/Eatery.py [hunk 1]
```diff
@@ -104,26 +112,35 @@ def to_json(
...
             "online_order_url": self.online_order_url,
-            "wait_times": None if self.wait_times is None else [wait_time.to_json() for wait_time in self.wait_times]
+            "wait_times": None if self.wait_times is None else [wait_time.to_json() for wait_time in self.wait_times],
+            "exceptions": None if self.exceptions is None else [exception.to_json() for exception in self.exceptions]
         }
         return eatery
 
+    @staticmethod 
+    def parse_field(json, key):
+        return None if key not in json else json[key]
```
Comment (whoiswillma) [inline api/datatype/Eatery.py]: does `json.get(key)` work here?

File: api/datatype/EateryException.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+class EateryException:
```
Comment (whoiswillma) [inline api/datatype/EateryException.py]: Wondering if there's a better name for these internally? Maybe EateryAlert?

File: api/datatype/Event.py [hunk 3]
```diff
@@ -50,9 +48,14 @@ def from_json(event_json):
     def __contains__(self, item: int):
         return self.start_timestamp <= item <= self.end_timestamp
 
-
-def _combined_timestamp(date: date, time: time, tzinfo: pytz.timezone) -> int:
-    return int(tzinfo.localize(datetime.combine(date, time)).timestamp())
+    #TODO: Restructure to move to some util file into major folder outside of api
```
Comment (whoiswillma) [inline api/datatype/Event.py]: Bump on this

File: api/datatype/MenuSubItem.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
+from typing import Optional
+
+class MenuSubItem:
```
Comment (whoiswillma) [inline api/datatype/MenuSubItem.py:3]: I would think the nesting would go:

`MenuItemSection` > `MenuItem` > `MenuSubItem`

Would it make sense to rename this to `MenuItem`?

File: api/dfg/schedule/ClosedSchedule.py [hunk 5]
```diff
@@ -1,15 +1,16 @@
 from api.dfg.DfgNode import DfgNode
 from api.datatype.Eatery import Eatery, EateryID
-from eateries.models import ClosedEventSchedule
+# from eateries.models import ClosedEventSchedule
```
Comment (whoiswillma) [inline api/dfg/schedule/ClosedSchedule.py]:
```suggestion
```

**Review by whoiswillma — APPROVED** [review]

File: api/datatype/Eatery.py (hunk 1 above)
Comment (connorreinhold) [reply api/datatype/Eatery.py]: RIP I needed this

**Pull request merged**